### DISCUSSION TITLE
[BUG] Wrong next run's time

### DISCUSSION THREAD

**Discussion by hungdoansy** [go-co-op/gocron#277] [Q&A] [answered]
Hi,

I tried to get time of the next run, but the function kept returning `next run 0001-01-01 00:00:00 +0000 UTC` no matter how the time schedule was. I tried both Cron and Every functions.

I tried this code on playground ([link](https://go.dev/play/p/UXzyWyzFG9w)) and it worked just fine. I have no idea what's wrong in my implemetation.

Below is the code I used in my macbook

```go
s := gocron.NewScheduler(time.UTC)

// Every 1 hour
job, err := s.Cron("0 * * * *").Do(func() {
logger.Info("starts")

// ... do something

logger.Info("ends")
})

t := job.NextRun()
logger.Infof("next run %s", t)
```

**Comment by JohnRoesler** [comment, marked as answer]
The scheduler doesn't calculate next run times until it has been started. Once you start the scheduler and call next run you'll get the right time! 

```golang
package main

import (
	"log"
	"time"

	"github.com/go-co-op/gocron"
)

func main() {
	s := gocron.NewScheduler(time.UTC)

	// Every 1 hour
	job, err := s.Cron("0 * * * *").Do(func() {
		log.Println("start")

		// ... do something

		log.Println("ends")
	})
	if err != nil {
		log.Fatalln(err)
	}

	t := job.NextRun()
	log.Printf("next run %s", t)

	s.StartAsync()

	t = job.NextRun()
	log.Printf("next run %s", t)
}
```

**Reply by hungdoansy** [reply]
Thank you so much
